epitome/models.py

```diff
@@ -746,3 +751,148 @@ def create_model(self, **kwargs):
         model = tf.keras.models.Model(inputs=cell_inputs, outputs=outputs)
 
         return model
+    
+    def score_matrix(self, accessibility_peak_matrix, regions):
+        """ Runs predictions on a matrix of accessibility peaks, where columns are samples and
+        rows are regions from regions_peak_file. rows in accessilibility_peak_matrix should matching
+
+        :param numpy.matrix accessilibility_peak_matrix:  of (samples by genomic regions)
+        :param str regions: either narrowpeak or bed file containing regions to score, OR a pyranges object
+            with columns [Chomosome, Start, End, idx]. Index matches each genomic region to a row in
+            accessilibility_peak_matrix. In both cases, number of regions Should
+            match rows in accessilibility_peak_matrix
+
+        :return: 3-dimensional numpy matrix of predictions: sized (samples by regions by ChIP-seq targets)
+        :rtype: numpy matrix
+        """
+
+        conversionObject = RegionConversion(self.dataset.regions, regions)
+
+        results = []
+        # print(accessibility_peak_matrix.shape)
+        matrix, indices = conversionObject.get_binary_vector(vector = accessibility_peak_matrix[0,:])
+        gen = load_data(data=self.dataset.get_data(Dataset.ALL),
+                 label_cell_types=self.test_celltypes,   # used for labels. Should be all for train/eval and subset for test
+                 eval_cell_types=self.eval_cell_types,   # used for rotating features. Should be all - test for train/eval
+                 matrix=self.dataset.matrix,
+                 targetmap=self.dataset.targetmap,
+                 cellmap=self.dataset.cellmap,
+                 radii = self.radii,
+                 mode = Dataset.RUNTIME,
+                 similarity_matrix = matrix,
+                 similarity_targets = ['DNase'],
+                 indices = indices,
+                 continuous=True,
+                 keep_lbl_mask=False,
+                 return_feature_names=False)
+
+        to_stack = load_data(data=self.dataset.get_data(Dataset.ALL),
+                 label_cell_types=self.test_celltypes,   # used for labels. Should be all for train/eval and subset for test
+                 eval_cell_types=self.eval_cell_types,   # used for rotating features. Should be all - test for train/eval
+                 matrix=self.dataset.matrix,
+                 targetmap=self.dataset.targetmap,
+                 cellmap=self.dataset.cellmap,
+                 radii = self.radii,
+                 mode = Dataset.RUNTIME,
+                 similarity_matrix = matrix,
+                 similarity_targets = ['DNase'],
+                 indices = indices,
+                 continuous=True,
+                 return_feature_names=True)
+
+        gen_to_list = list(gen())
+        to_stack = list(to_stack())
+        gen_to_list = np.array(gen_to_list)
```

Comment:
doesn't this still have the labels and masks in it? so each item in the array is a tuple? how does this work when you pass it into the CASV?

Reply:
`to_stack` isn't passed into the CASV. I added a parameter in `load_data` (`keep_lbl_mask`) which if set to True, returns the labels and masks. For `gen_to_list`, this flag is set to false so it doesn't have the labels or masks.